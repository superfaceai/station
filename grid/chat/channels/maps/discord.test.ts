import { SuperfaceTest } from '@superfaceai/testing';

/**
 * Live tests
 *
 * @group live/safe
 */

describe('chat/channels/discord', () => {
  let superface: SuperfaceTest;

  describe('GetChannels', () => {
    beforeAll(() => {
      superface = new SuperfaceTest({
        profile: 'chat/channels',
        provider: 'discord',
        useCase: 'GetChannels',
        testInstance: expect,
      });
    });

    describe('when specified destination does exist', () => {
      it('performs correctly', async () => {
        const result = await superface.run({
          input: {
            workspace: '935962220104396881',
            visibility: 'public',
          },
        });

        expect(result).toMatchSnapshot({
          value: {
            channels: expect.arrayContaining([
              expect.objectContaining({
                createdAt: expect.any(Number),
                id: expect.any(String),
                name: 'announcements',
              }),
              expect.objectContaining({
                createdAt: expect.any(Number),
                id: expect.any(String),
                name: 'resources',
              }),
              expect.objectContaining({
                createdAt: expect.any(Number),
                id: expect.any(String),
                name: 'general',
              }),
            ]),
            rateLimit: expect.objectContaining({
              bucket: expect.any(String),
              remainingRequests: expect.any(Number),
              remainingRequestsPercentage: expect.any(Number),
              resetAfter: expect.any(Number),
              resetTimestamp: expect.any(Number),
              totalRequests: expect.any(Number),
            }),
          },
        });
      });
    });

    describe('when specified destination does not exist', () => {
      it('returns error', async () => {
        const result = await superface.run(
          {
            input: {
              workspace: '000000000000000000',
              visibility: 'public',
            },
          },
          { fullError: true }
        );

        expect(result).toMatchSnapshot({
          error: expect.objectContaining({
            kind: expect.stringMatching('HTTPError'),
            message: expect.stringMatching('Expected HTTP error'),
            properties: expect.objectContaining({
              title: 'Not found',
              detail: 'Unknown Guild',
              rateLimit: expect.objectContaining({
                bucket: expect.any(String),
                remainingRequests: expect.any(Number),
                remainingRequestsPercentage: expect.any(Number),
                resetAfter: expect.any(Number),
                resetTimestamp: expect.any(Number),
                totalRequests: expect.any(Number),
              }),
            }),
            statusCode: expect.any(Number),
          }),
        });
      });
    });
  });
});
