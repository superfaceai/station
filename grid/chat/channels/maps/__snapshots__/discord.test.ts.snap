// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat/channels/discord GetChannels when specified destination does exist performs correctly 1`] = `
Object {
  "value": Object {
    "channels": ArrayContaining [
      ObjectContaining {
        "createdAt": Any<Number>,
        "id": Any<String>,
        "name": "announcements",
      },
      ObjectContaining {
        "createdAt": Any<Number>,
        "id": Any<String>,
        "name": "resources",
      },
      ObjectContaining {
        "createdAt": Any<Number>,
        "id": Any<String>,
        "name": "general",
      },
    ],
    "rateLimit": ObjectContaining {
      "bucket": Any<String>,
      "remainingRequests": Any<Number>,
      "remainingRequestsPercentage": Any<Number>,
      "resetAfter": Any<Number>,
      "resetTimestamp": Any<Number>,
      "totalRequests": Any<Number>,
    },
  },
}
`;

exports[`chat/channels/discord GetChannels when specified destination does not exist returns error 1`] = `
Object {
  "error": ObjectContaining {
    "kind": StringMatching /HTTPError/,
    "message": StringMatching /Expected HTTP error/,
    "properties": ObjectContaining {
      "detail": "Unknown Guild",
      "rateLimit": ObjectContaining {
        "bucket": Any<String>,
        "remainingRequests": Any<Number>,
        "remainingRequestsPercentage": Any<Number>,
        "resetAfter": Any<Number>,
        "resetTimestamp": Any<Number>,
        "totalRequests": Any<Number>,
      },
      "title": "Not found",
    },
    "statusCode": Any<Number>,
  },
}
`;
